vllm/model_executor/layers/quantization/kernels/scaled_mm/__init__.py

```diff
@@ -0,0 +1,79 @@
+import os
+from typing import Dict, List, Optional, Type
+
+from vllm.model_executor.layers.quantization.kernels.scaled_mm.cutlass import (
+    CutlassScaledMMLinearKernel)
+from vllm.model_executor.layers.quantization.kernels.scaled_mm.ScaledMMLinearKernel import (  # noqa: E501
+    ScaledMMLinearKernel, ScaledMMLinearLayerConfig)
+from vllm.model_executor.layers.quantization.kernels.scaled_mm.xla import (
+    XLAScaledMMLinearKernel)
+from vllm.platforms import PlatformEnum, current_platform
+
+# in priority/performance order (when available)
+_POSSIBLE_KERNELS: Dict[PlatformEnum, List[Type[ScaledMMLinearKernel]]] = {
+    PlatformEnum.CPU: [CutlassScaledMMLinearKernel],
+    PlatformEnum.CUDA: [CutlassScaledMMLinearKernel],
```

Comment:
I believe `PlatformEnum.ROCM` should be here as well since there is INT8 W8A8 support https://github.com/vllm-project/vllm/pull/9857

Reply:
good catch. this abstraction is much better

Reply:
NOTE: to keep the scope of this PR reasonable, I will add `TritonScaledMM` Kernel as a follow up.